src/main/java/art/heredium/service/CouponUsageService.java

```diff
@@ -41,4 +46,40 @@ public List<CouponResponseDto> getCouponsWithUsageByAccountId(Long accountId) {
 
     return responseDtos;
   }
+
+  @Transactional(rollbackFor = Exception.class)
+  public List<CouponUsage> distributeCoupons(
+      @NonNull Account account, @NonNull List<Coupon> coupons) {
+    LocalDateTime distributionDateTime = LocalDateTime.now();
+    List<CouponUsage> couponUsages = new ArrayList<>();
+    coupons.forEach(
+        coupon -> {
+          long numberOfUses = Optional.ofNullable(coupon.getNumberOfUses()).orElse(1L);
+          boolean isPermanentCoupon = Boolean.TRUE.equals(coupon.getIsPermanent());
+          if (isPermanentCoupon) {
+            CouponUsage couponUsage =
+                new CouponUsage(
+                    coupon,
+                    account,
+                    distributionDateTime,
+                    distributionDateTime.plusDays(coupon.getPeriodInDays()),
+                    true,
+                    0L);
+            couponUsages.add(couponUsage);
+          } else {
+            for (int i = 0; i < numberOfUses; i++) {
+              CouponUsage couponUsage =
+                  new CouponUsage(
+                      coupon,
+                      account,
+                      distributionDateTime,
+                      distributionDateTime.plusDays(coupon.getPeriodInDays()),
+                      false,
+                      null);
```

Comment:
```suggestion
                      0);
```
Cho dù là coupon ko phải permanent nhưng vẫn nên để numberOfUse là 0 để thống nhất các loại coupon. Khi coupon này dc dùng thì mình sẽ update cả 2 trường (isUsed = true và numberOfUses = 1)

Reply:
Đồng ý, tôi đã fix: 720a0e166d319548317a68a3a9d34d112a387eae